lib/src/algorithms.dart

```diff
@@ -247,7 +247,7 @@ void _merge<T>(
     List<T> firstList,
     int firstStart,
     int firstEnd,
-    List<T?> secondList,
+    List<T> secondList,
```

Comment:
I worry the `cast` is expensive. Why did this need to change?

Reply:
@lrhn do you want to take a look at this?

The problem is that `_merge` calls `target.setRange` with this List. The `target` parameter has a type of `List<T?>` but we actually pass in a `List<T>`, so the `setRange` call fails.

I am not sure what the most efficient way to fix this code is going to be, this version adds a lot of extra casting.

I think for now we can just land this fix but it would be good to optimize it.